app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -139,6 +140,14 @@ open class HomeActivity : LocaleAwareAppCompatActivity() {
         hotStartMonitor.onPostResumeFinalMethodCall()
     }
 
+    final override fun onPause() {
+        super.onPause()
+
+        // Every time the application goes into the background, it is possible that the user
+        // is about to change the browsers installed on their system. Therefore, we reset the cache of
+        // all the installed browsers.
+        BrowsersCache.resetAll()
```

Comment:
It's also possible for the installed browsers to change while the app is still running. You can install an app to your phone from the Google Play website, and it will get installed in the background.

Reply:
@hawkinsw , @NotWoods Maybe we could use a `BroadcastReceiver` for `ACTION_PACKAGE_ADDED` and `ACTION_PACKAGE_REMOVED`   ?